mbed_build/mbed_build.py

```diff
@@ -4,25 +4,29 @@
 #
 """Module to generate application's CMake file."""
 import pathlib
-from typing import List
-
+from typing import Iterable
+from mbed_targets.mbed_target_build_attributes import MbedTargetBuildAttributes
 from mbed_build._internal.cmake_file import render_cmakelists_template, write_cmakelists_file
 from mbed_build.exceptions import InvalidExportOutputDirectory
 
 
-def generate_cmakelists_file(target_name: str, toolchain_name: str) -> str:
+def generate_cmakelists_file(target_build_attributes: MbedTargetBuildAttributes, toolchain_name: str) -> str:
```

Comment:
Now that I see how this is used, I'm not sure what the purpose of `generate_` vs cli command is. Should the cli command be responsible for assembling data, or should it forward given arguments to `generate_cmakelists_file`? Currently it seems that data assembly would happen in both places.

Reply:
My experience is that _heavy_ logic should happen in function that is called by the cli command. I.e.: make cli command only responsible for parsing arguments and deciding which path to use, everything else should happen downstream.

Reply:
Yeah I see your point. Maybe I should just move the generate stuff into the cli command. Its basically just coordinating which seems top-level.

Reply:
Any is fine tbh. 
I was suggesting pushing it down, as testing click cli gets muddy the moment you have more than one codepath.

Reply:
Yeah you are right. I was trying to keep to the JCM command of pushing contact to the outside world into the 'edges' of the code. But I think you are right and I am applying it badly here. I'll change it.

Reply:
3ed2597